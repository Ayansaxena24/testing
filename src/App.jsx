import { useState } from 'react'
import React, { Component } from 'react'
// import './App.css'

class App extends Component {
  state = {
    count: 0
  }

  makeIncrementer = amount => () => {
  this.setState(prevState => ({
      count : prevState.count + amount,
  })
  )
}

increment = this.makeIncrementer(1)
decrement = this.makeIncrementer(-1)

  // increment = () =>
  //   this.setState(prevState => {
  //     return {
  //       count: prevState.count + 1,
  //     }
  //   })

  //   decrement = () =>
  //   this.setState(prevState => {
  //     return {
  //       count: prevState.count - 1,
  //     }
  //   })

    render() {
      return (
        <div className='card'>
          <button className='increment' onClick={this.increment}>+++</button>
          <p>{this.state.count}</p>
          <button className='decrement' onClick={this.decrement}>---</button>
        </div>
      )
    }
}

export default App
