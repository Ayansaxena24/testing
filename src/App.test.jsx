import React from "react";
import App from "./App";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import renderer from "react-test-renderer";

configure({ adapter: new Adapter() });

describe("Counter Component", () => {
  it("Matches the snapshot", () => {
    const tree = renderer.create(<App />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("starts with a count of 0", () => {
    const wrapper = shallow(<App />);
    const countState = wrapper.state().count;
    expect(countState).toEqual(0);
  });
  it("increments count by 1 when the increment button is clicked", () => {
    const wrapper = shallow(<App />);
    const incrementBtn = wrapper.find("button.increment");
    incrementBtn.simulate("click");
    const text = wrapper.find("p").text();
    expect(text).toEqual("1");
  });
  it("decrements count by 1 when the decrement button is clicked", () => {
    const wrapper = shallow(<App />);
    const decrementBtn = wrapper.find("button.decrement");
    decrementBtn.simulate("click");
    const text = wrapper.find("p").text();
    expect(text).toEqual("-1");
  });
});
