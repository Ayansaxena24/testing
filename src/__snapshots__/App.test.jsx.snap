// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Matches the snapshot 1`] = `
<div
  className="card"
>
  <button
    className="increment"
    onClick={[Function]}
  >
    +++
  </button>
  <p>
    0
  </p>
  <button
    className="decrement"
    onClick={[Function]}
  >
    ---
  </button>
</div>
`;
